test/pytest/test_keras_api.py

```diff
@@ -297,6 +299,58 @@ def test_conv2d(chans, padds, backend, io_type):
         assert list(hls_model.get_layers())[1].attributes['pad_right'] == 0
 
 
+# Currently only Vivado and Vitis is supported for io_stream.
+@pytest.mark.parametrize('backend', ['Vivado', 'Vitis'])
+@pytest.mark.parametrize('io_type', ['io_stream'])
+def test_depthwise2d(backend, io_type):
+    '''
+    Test proper handling of DepthwiseConv2D
+    '''
+    X = np.random.rand(10, 32, 32, 3)
+    X = np.round(X * 2**10) * 2**-10  # make it an exact ap_fixed<16,6>
+    model = tf.keras.models.Sequential()
+    model.add(DepthwiseConv2D(kernel_size=(3, 3), input_shape=(32, 32, 3)))
+    model.compile()
+
+    config = hls4ml.utils.config_from_keras_model(model, granularity='name', default_precision='fixed<32,12>')
```

Comment:
If the input is clipped to exactly match <16,6>, why do we need to increase the precision this way? Should only `accum` be increased?

Reply:
It's not clipped for the keras input, just the hls4ml input. The input is actually all in the [0,1) range. I didn't try to optimize the precision in the model and just sought to make it irrelevant.